src/System.Management.Automation/engine/parser/Parser.cs

```diff
@@ -5692,7 +5703,212 @@ private StatementAst DataStatementRule(Token dataToken)
+                        currentPipelineChain,
+                        new ErrorStatementAst(errorPosition),
+                        currentChainOperatorToken.Kind);
+                }
+
+                // Look ahead for a chain operator
+                nextToken = PeekToken();
+                switch (nextToken.Kind)
+                {
+                    case TokenKind.AndAnd:
+                    case TokenKind.OrOr:
+                        SkipToken();
+                        SkipNewlines();
+                        break;
+
+                    // Background operators may also occur here
+                    case TokenKind.Ampersand:
+                        SkipToken();
+                        nextToken = PeekToken();
+
+                        switch (nextToken.Kind)
+                        {
+                            case TokenKind.AndAnd:
+                            case TokenKind.OrOr:
+                                SkipToken();
+                                ReportError(nextToken.Extent, nameof(ParserStrings.BackgroundOperatorInPipelineChain), ParserStrings.BackgroundOperatorInPipelineChain);
+                                return new ErrorStatementAst(ExtentOf(currentPipelineChain ?? nextPipeline, nextToken.Extent));
+                        }
+
+                        background = true;
+                        goto default;
+
+                    // No more chain operators -- return
+                    default:
+                        // If we haven't seen a chain yet, pass through the pipeline
+                        // Simplifies the AST and prevents allocation
+                        if (currentPipelineChain == null)
+                        {
+                            if (!background)
+                            {
+                                return nextPipeline;
+                            }
+
+                            // Set background on the pipeline AST
+                            nextPipeline.Background = true;
+                            return nextPipeline;
+                        }
+
+                        return new PipelineChainAst(
+                            ExtentOf(currentPipelineChain.Extent, nextPipeline.Extent),
+                            currentPipelineChain,
+                            nextPipeline,
+                            currentChainOperatorToken.Kind,
+                            background);
+                }
+
+                // Assemble the new chain statement AST
+                ChainableAst nextPipelineChain = nextPipeline;
+                currentPipelineChain = currentPipelineChain == null
+                    ? nextPipelineChain
+                    : new PipelineChainAst(
+                        ExtentOf(currentPipelineChain.Extent, nextPipeline.Extent),
+                        currentPipelineChain,
+                        nextPipelineChain,
+                        currentChainOperatorToken.Kind);
```

Comment:
This generates a left-hand-side heavy tree, while the compiler is expecting a right-hand-side heavy tree.

Reply:
Refactored here and in the compiler to correct this